src/client/app/redux/api/readingsApi.ts

```diff
@@ -0,0 +1,156 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as _ from 'lodash';
+import {
+	BarReadingApiArgs,
+	CompareReadingApiArgs,
+	LineReadingApiArgs,
+	RadarReadingApiArgs,
+	ThreeDReadingApiArgs
+} from '../selectors/chartQuerySelectors';
+import { RootState } from '../../store';
+import { BarReadings, CompareReadings, LineReadings, ThreeDReading } from '../../types/readings';
+import { baseApi } from './baseApi';
+
+export const readingsApi = baseApi.injectEndpoints({
+	endpoints: builder => ({
+		threeD: builder.query<ThreeDReading, ThreeDReadingApiArgs>({
+			// ThreeD requests only single meters at a time which plays well with default cache behavior
+			// No other properties are necessary for this endpoint
+			// Refer to the line endpoint for an example of an endpoint with custom cache behavior
+			query: ({ id, timeInterval, graphicUnitId, readingInterval, meterOrGroup }) => ({
+				// destructure args that are passed into the callback, and generate/ implicitly return the API url for the request.
+				url: `api/unitReadings/threeD/${meterOrGroup}/${id}?`,
```

Comment:
Should there be a ? at the end (below does not have)?

Reply:
Should there be a ? at the

Reply:
Interesting that it works either way. I'm assuming the fetchBaseQuery is smart enough to sort itself out. I guess it should be omitted since it handles the url generation internally when given a url + params object.

Reply:
Removed.